src/main/groovy/edu/oregonstate/mist/personsapi/PersonsResource.groovy

```diff
@@ -0,0 +1,127 @@
+package edu.oregonstate.mist.personsapi
+
+import com.codahale.metrics.annotation.Timed
+import edu.oregonstate.mist.api.Resource
+import edu.oregonstate.mist.api.jsonapi.ResourceObject
+import edu.oregonstate.mist.api.jsonapi.ResultObject
+import edu.oregonstate.mist.personsapi.db.PersonsDAO
+import groovy.transform.TypeChecked
+
+import javax.annotation.security.PermitAll
+import javax.imageio.ImageIO
+import javax.ws.rs.GET
+import javax.ws.rs.Path
+import javax.ws.rs.PathParam
+import javax.ws.rs.Produces
+import javax.ws.rs.QueryParam
+import javax.ws.rs.core.MediaType
+import javax.ws.rs.core.Response
+import java.awt.image.BufferedImage
+
+@Path("persons")
+@Produces(MediaType.APPLICATION_JSON)
+@PermitAll
+@TypeChecked
+class PersonsResource extends Resource {
+    private final PersonsDAO personsDAO
+    private PersonUriBuilder personUriBuilder
+    private final Integer maxImageWidth = 2000
+
+    PersonsResource(PersonsDAO personsDAO, URI endpointUri) {
+        this.personsDAO = personsDAO
+        this.endpointUri = endpointUri
+        this.personUriBuilder = new PersonUriBuilder(endpointUri)
+    }
+
+    @Timed
+    @GET
+    Response list(@QueryParam('onid') String onid,
+                  @QueryParam('osuID') String osuID,
+                  @QueryParam('osuUID') String osuUID) {
+
+        def id = [onid, osuID, osuUID].findAll { it }
+        if (id.size() != 1) {
+            badRequest('The number of input parameter(s) is not equal to one.').build()
```

Comment:
I think we should make this error message more specific. We might be adding more query parameters to this API - and those parameters might be allowed to be included together. We could say something like "onid, osuID, and osuUID cannot not be included together"

Reply:
Good point.